src/commands/list_folder.ts

```diff
@@ -0,0 +1,130 @@
+		const wallet = await context.getWallet().catch(() => null);
+		const folderId = context.getParameterValue(ParentFolderIdParameter);
+		let folder;
+		let mergedData: (ArFSPrivateFileOrFolderData | ArFSPublicFileOrFolderData)[];
+
+		if (!folderId) {
+			console.log(`Folder id not specified! ${folderId}`);
+			process.exit(1);
+		}
+
+		if (wallet) {
+			const arDrive = new ArDrive(
+				wallet,
+				cliWalletDao,
+				new ArFSDAO(wallet, arweave),
+				new ArDriveCommunityOracle(arweave),
+				CLI_APP_NAME,
+				CLI_APP_VERSION
+			);
+			// Fetch the folder to extract the drive
+			folder = await arDrive.getPrivateFolder(folderId);
+
+			// Fetch all of the folder entities within the drive
+			const driveIdOfFolder = folder.driveId;
+			const allFolderEntitiesOfDrive = await arDrive.getAllFoldersOfPrivateDrive(driveIdOfFolder);
+
+			// Feed entities to FolderHierarchy.setupNodesWithEntity()
+			const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+			const folderIDs = hierarchy.allFolderIDs();
+
+			// Fetch all file entities within all Folders of the drive
+			const allFileEntitiesOfDrive = await arDrive.getPrivateChildrenFilesFromFolderIDs(folderIDs);
+
+			// Fetch all names of each entity
+			const allEntitiesOfDrive = [...allFolderEntitiesOfDrive, ...allFileEntitiesOfDrive].sort(
+				(a, b) => +a.txId - +b.txId
+			);
+
+			mergedData = allEntitiesOfDrive.map((entity) => {
+				const path = `${hierarchy.pathToFolderId(entity.parentFolderId)}/${entity.name}`;
+				const txPath = `${hierarchy.txPathToFolderId(entity.parentFolderId)}/${entity.txId}`;
+				const entityIdPath = `${hierarchy.entityPathToFolderId(entity.parentFolderId)}/${entity.entityId}`;
+				return new ArFSPrivateFileOrFolderData(
+					entity.appName,
+					entity.appVersion,
+					entity.arFS,
+					entity.contentType,
+					entity.driveId,
+					entity.entityType,
+					entity.name,
+					entity.txId,
+					entity.unixTime,
+					entity.parentFolderId,
+					entity.entityId,
+					entity.cipher,
+					entity.cipherIV,
+					path,
+					txPath,
+					entityIdPath
+				);
+			});
+		} else {
+			const arDrive = new ArDriveAnonymous(new ArFSDAOAnonymous(arweave));
+			folder = await arDrive.getPublicFolder(folderId);
+
+			// Fetch all of the folder entities within the drive
+			const driveIdOfFolder = folder.driveId;
+			const allFolderEntitiesOfDrive = await arDrive.getAllFoldersOfPublicDrive(driveIdOfFolder);
+
+			// Feed entities to FolderHierarchy.setupNodesWithEntity()
+			const hierarchy = FolderHierarchy.newFromEntities(allFolderEntitiesOfDrive);
+			const folderIDs = hierarchy.allFolderIDs();
+
+			// Fetch all file entities within all Folders of the drive
+			const allFileEntitiesOfDrive = await arDrive.getPublicChildrenFilesFromFolderIDs(folderIDs);
+
+			// Fetch all names of each entity
+			const allEntitiesOfDrive = [...allFolderEntitiesOfDrive, ...allFileEntitiesOfDrive].sort(
```

Comment:
Omit sort. Sort after the merge on path.

Reply:
Now sorted by path: e8735f640d062afd36947ba650b39e4b1720d9f1